### PR TITLE
add type annotations to io\formats\html.py

### PR DESCRIPTION
**Pull request by simonjayhawkins**

### CODE REVIEW

File: pandas/io/formats/html.py [hunk 1]
```diff
@@ -111,20 +123,24 @@ def write_th(self, s, header=False, indent=0, tags=None):
...
             start_tag = "<{kind} {tags}>".format(kind=kind, tags=tags)
         else:
             start_tag = "<{kind}>".format(kind=kind)
 
         if self.escape:
             # escape & first to prevent double escaping of &
-            esc = OrderedDict([("&", r"&amp;"), ("<", r"&lt;"), (">", r"&gt;")])
+            esc = OrderedDict(
+                [("&", r"&amp;"), ("<", r"&lt;"), (">", r"&gt;")]
+            )  # type: Union[OrderedDict[str, str], Dict]
```
Comment (WillAyd) [inline pandas/io/formats/html.py]: I think also OK to just make this an ordereddict all of the time; simplifies annotation and code

**Review by WillAyd — APPROVED** [review]
Nice job on these. Optional comment but otherwise lgtm

**Comment by jreback** [comment]
thanks @simonjayhawkins really nice!

File: pandas/io/formats/html.py (hunk 1 above)
Comment (simonjayhawkins) [reply pandas/io/formats/html.py]: i was getting `pandas\io\formats\html.py:145: error: Incompatible types in assignment (expression has type "Dict[<nothing>, <nothing>]", variable has type "OrderedDict[str, str]")`

Comment (WillAyd) [reply pandas/io/formats/html.py]: Yea I so meant you *could* change the dict on line 145 to be an OrderedDict to match the signature here. I think OK to refactor code if it simplifies annotations and keeps same functionality, as there probably helps readability too

Though this is obviously fine as is too!

**Pull request merged**